src/CoreServices/FSEvents.cs

```diff
@@ -307,6 +331,31 @@ public void ScheduleWithRunLoop (NSRunLoop runLoop)
 			ScheduleWithRunLoop (runLoop.GetCFRunLoop (), CFRunLoop.ModeDefault);
 		}
 
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern void FSEventStreamUnscheduleFromRunLoop (IntPtr handle,
+			IntPtr runLoop, IntPtr runLoopMode);
+
+		public void UnscheduleWithRunLoop (CFRunLoop runLoop, NSString runLoopMode)
+		{
+			CheckDisposed ();
+			FSEventStreamUnscheduleFromRunLoop (handle, runLoop.Handle, runLoopMode.Handle);
```

Comment:
you'll get `NRE` if other arguments are `null`

If null is allowed then use the `GetHandle` extension method
if not, then add a null check and throw an `ANE`